### PR TITLE
Singleton neo4j

### PR DESCRIPTION
**Pull request by jodom**
A singleton neo4jService that encapsulates driver. Layers don't need to have a knowledge of underlying neo4j driver. Composition proposed here allows us to smoothly swap out drivers in future #65, and remove driver instantiation on critical boot path #78.
Additionally, this PR formalizes a result API on neo4jService endpoints. Success from service calls return as `{ status: 200, result: someResult }` while failures return `{ status: 500, error: errorWithStack }`

Next: This pattern opens up an opportunity for better error ux, most immediately, a path for solving #77 [ don't rerender if we hit error on create/update new layer ]

PS: Builds on top of changes in PR #75

### CODE REVIEW

**Comment by jodom** [comment]
cc @wagnerjt for reviews on service changes

**Comment by jodom** [comment]
Not ready until I add tests for the driver... :)

**Comment by jodom** [comment]
> Not ready until I add tests for the driver... :)

done. @stellasia, @wagnerjt I'd love your reviews on pr

**Comment by jodom** [comment]
> Not ready until I add tests for the driver... :)

resolved. Tests covering singleton driver pattern are in place

**Comment by jodom** [comment]
> A singleton neo4jService that encapsulates driver. Layers don't need to have a knowledge of underlying neo4j driver. Composition proposed here allows us to smoothly swap out drivers in future #65, and remove driver instantiation on critical boot path #78.
> Additionally, this PR formalizes a result API on neo4jService endpoints. Success from service calls return as `{ status: 200, result: someResult }` while failures return `{ status: 500, error: errorWithStack }`
> 
> Next: This pattern opens up an opportunity for better error ux, most immediately, a path for solving #77 [ don't rerender if we hit error on create/update new layer ]
> 
> PS: Builds on top of changes in PR #75

@stellasia, let's review and close this PR when possible.

**Comment by stellasia** [comment]
Sure, will do during the coming week!

**Review by stellasia — CHANGES_REQUESTED** [review]
I found a couple of other bugs when reviewing but not related to this PR. So looks good to me when the issue with `records` is fixed (see comment in code)

File: src/services/neo4jService.js [hunk 1]
```diff
@@ -1,185 +1,175 @@
...
+    const driver = this.driver || (await this._getNeo4jDriver());
+
+    if (!driver) {
+      throw new Error("Failed to get driver");
+    }
+
+    const session = driver.session();
+
+    const records = await session.run(query, params).then(
+      (records) => records,
```
[the remark this answers is not in the document]
Comment (jodom) [reply src/services/neo4jService.js]: Oh. I even wrote UTs reflecting a flatter result object. Fixed!

**Review by stellasia — APPROVED** [review]

**Comment by stellasia** [comment]
Thank you very much for this @jodom , merging :100:

**Pull request merged**